In [23]:
import pandas as pd

In [24]:
variable_dict_df = pd.read_csv('Data/American Community Survey Data/acs5_variable_dict_2014_2019.csv')

In [25]:
variable_dict_df.head()

variable_code                                              label  \
0   DP02_0019EA  Annotation of Estimate!!RELATIONSHIP!!Populati...   
1    DP02_0126E       Estimate!!ANCESTRY!!Total population!!Danish   
2   DP02_0072EA  Annotation of Estimate!!DISABILITY STATUS OF T...   
3  DP02_0069PMA  Annotation of Percent Margin of Error!!VETERAN...   
4    DP02_0126M  Margin of Error!!ANCESTRY!!Total population!!D...   

                                             concept predicateType group  \
0  SELECTED SOCIAL CHARACTERISTICS IN THE UNITED ...        string  DP02   
1  SELECTED SOCIAL CHARACTERISTICS IN THE UNITED ...           int  DP02   
2  SELECTED SOCIAL CHARACTERISTICS IN THE UNITED ...        string  DP02   
3  SELECTED SOCIAL CHARACTERISTICS IN THE UNITED ...        string  DP02   
4  SELECTED SOCIAL CHARACTERISTICS IN THE UNITED ...           int  DP02   

   limit  predicateOnly  acs_year  
0      0           True      2014  
1      0           True      2014  
2      0           True      2014  
3      0           True      2014  
4      0           True      2014

In [26]:
variable_dict_df.dtypes

variable_code    object
label            object
concept          object
predicateType    object
group            object
limit             int64
predicateOnly      bool
acs_year          int64
dtype: object

In [27]:
#Count the No. of variable codes in 2014 and 2019
variable_dict_df.groupby('acs_year')['variable_code'].count()

acs_year
2014    4088
2019    4176
Name: variable_code, dtype: int64

There are more variable codes in 2019 than in 2014. To be exact, there are 88 extra codes in 2019

In [28]:
# Check what variable_codes appear in 2014 set alone or 2019 set alone
variable_dict_2014 = variable_dict_df.loc[variable_dict_df['acs_year']==2014]
variable_dict_2019 = variable_dict_df.loc[variable_dict_df['acs_year']==2019]

In [29]:
# Find variable codes that appear in 2014 but not in 2019
variable_code_2019 = variable_dict_2019['variable_code'].tolist()
for variable_code in variable_dict_2014['variable_code']:
    if variable_code not in variable_code_2019:
        print(variable_code)

All varible codes that appear in 2014 also appear in 2019

In [30]:
# Find variable codes that appear in 2019 but not in 2014
count_codes = 0
variable_code_2014 = variable_dict_2014['variable_code'].tolist()
for variable_code in variable_dict_2019['variable_code']:
    if variable_code not in variable_code_2014:
        count_codes +=1
        print(variable_code)
        
print ('The No of variable codes that appear in 2019 but not in 2014: {}'.format(count_codes))

DP02_0153PEA
DP02_0153PMA
DP02_0153EA
DP02_0153MA
DP02_0153PE
DP02_0153PM
DP02_0153M
DP02_0153E
DP04_0143PEA
DP04_0143PMA
DP04_0142EA
DP04_0142MA
DP04_0142PM
DP04_0142PE
DP04_0143EA
DP04_0143MA
DP04_0143PM
DP04_0143PE
DP04_0142PEA
DP04_0142PMA
DP04_0143M
DP04_0143E
DP04_0142M
DP04_0142E
DP05_0089PMA
DP05_0089PEA
DP05_0082EA
DP05_0082MA
DP05_0082PM
DP05_0082PE
DP05_0083EA
DP05_0084PMA
DP05_0084PEA
DP05_0088EA
DP05_0088M
DP05_0088E
DP05_0087M
DP05_0089M
DP05_0089E
DP05_0087PMA
DP05_0084E
DP05_0083M
DP05_0085E
DP05_0084M
DP05_0088MA
DP05_0086E
DP05_0088PM
DP05_0085M
DP05_0088PE
DP05_0087E
DP05_0086M
DP05_0087PEA
DP05_0082E
DP05_0083E
DP05_0082M
DP05_0089EA
DP05_0089MA
DP05_0089PM
DP05_0089PE
DP05_0085PM
DP05_0085PE
DP05_0082PEA
DP05_0082PMA
DP05_0085PMA
DP05_0086EA
DP05_0085PEA
DP05_0086MA
DP05_0086PM
DP05_0086PE
DP05_0088PMA
DP05_0088PEA
DP05_0087EA
DP05_0087MA
DP05_0087PM
DP05_0087PE
DP05_0083MA
DP05_0083PM
DP05_0083PE
DP05_0083PMA
DP05_0084EA
DP05_0084MA
DP05_0083PEA
DP05_0084PM
DP05_0

The No of variable codes that appear in 2019 but not in 2014: 88

In [31]:
#count the variable codes whose labels are different between 2014 and 2019
count_mismatched_labels = 0        
for var_code in variable_code_2014:
    label_2014_series = variable_dict_2014.loc[variable_dict_2014['variable_code']==var_code]['label']
    label_2014 = label_2014_series[label_2014_series.index[0]] #since only one entry is available for each variable code in agiven year
    label_2019_series = variable_dict_2019.loc[variable_dict_2019['variable_code']==var_code]['label']
    label_2019 = label_2019_series[label_2019_series.index[0]] #since only one entry is available for each variable code in agiven year
    if  label_2014 !=  label_2019:
        count_mismatched_labels +=1
        print(var_code)

DP02_0019EA
DP02_0126E
DP02_0072EA
DP02_0069PMA
DP02_0126M
DP02_0057PMA
DP02_0125M
DP02_0054PM
DP02_0041MA
DP02_0070PMA
DP02_0125E
DP02_0054PE
DP02_0108MA
DP02_0124M
DP02_0124E
DP02_0098MA
DP02_0082PMA
DP02_0006EA
DP02_0139EA
DP02_0045PMA
DP02_0123M
DP02_0123E
DP02_0122M
DP02_0082PEA
DP02_0122E
DP02_0143PE
DP02_0130MA
DP02_0143PM
DP02_0121M
DP02_0085EA
DP02_0010PE
DP02_0121E
DP02_0010PM
DP02_0130PE
DP02_0130PM
DP02_0120M
DP02_0067PM
DP02_0120E
DP02_0054MA
DP02_0067PE
DP02_0094PEA
DP02_0123PMA
DP02_0085PM
DP02_0132PEA
DP02_0072MA
DP02_0085PE
DP02_0019MA
DP02_0113EA
DP02_0005PEA
DP02_0144PEA
DP02_0037EA
DP02_0100EA
DP02_0111PMA
DP02_0090EA
DP02_0019PM
DP02_0139MA
DP02_0017PEA
DP02_0072PM
DP02_0072PE
DP02_0019PE
DP02_0006MA
DP02_0094PMA
DP02_0041PE
DP02_0126EA
DP02_0041PM
DP02_0042PEA
DP02_0129E
DP02_0098PM
DP02_0129M
DP02_0085MA
DP02_0098PE
DP02_0128E
DP02_0128M
DP02_0029PEA
DP02_0127E
DP02_0108PE
DP02_0030PEA
DP02_0108PM
DP02_0127M
DP02_0123PEA
DP02_0084M
DP02_0085M
DP02_0132PMA
DP02_00

DP02_0106PEA
DP02_0029E
DP02_0020PE
DP02_0095EA
DP02_0046PMA
DP02_0105EA
DP02_0029M
DP02_0020PM
DP02_0028E
DP02_0095PEA
DP02_0077PM
DP02_0028M
DP02_0064MA
DP02_0077PE
DP02_0027E
DP02_0027M
DP02_0029EA
DP02_0140PE
DP02_0034PMA
DP02_0140PM
DP02_0026M
DP02_0026E
DP02_0082EA
DP02_0051EA
DP02_0025M
DP02_0025E
DP02_0124PMA
DP02_0071PEA
DP02_0033PE
DP02_0020MA
DP02_0131PEA
DP02_0024M
DP02_0033PM
DP02_0024E
DP02_0058PMA
DP02_0140MA
DP02_0023M
DP02_0077MA
DP02_0023E
DP02_0143PEA
DP02_0022M
DP02_0118EA
DP02_0022E
DP02_0083PEA
DP02_0112PMA
DP02_0118PE
DP02_0100PMA
DP02_0105MA
DP02_0118PM
DP02_0095MA
DP02_0003EA
DP02_0136EA
DP02_0095PMA
DP02_0082MA
DP02_0123EA
DP02_0006PEA
DP02_0095PM
DP02_0095PE
DP02_0029MA
DP02_0018PEA
DP02_0064PM
DP02_0071PMA
DP02_0149EA
DP02_0051MA
DP02_0064PE
DP02_0124PEA
DP02_0131PMA
DP02_0083PMA
DP02_0016EA
DP02_0118MA
DP02_0143PMA
DP02_0051PM
DP02_0112PEA
DP02_0051PE
DP02_0100PEA
DP02_0034EA
DP02_0098PMA
DP02_0016PM
DP02_0149PE
DP02_0136MA
DP02_0149PM
DP02_0016PE
DP02_0003

DP03_0132E
DP03_0118MA
DP03_0132M
DP03_0082MA
DP03_0130E
DP03_0130M
DP03_0136EA
DP03_0068PEA
DP03_0093PMA
DP03_0123MA
DP03_0081PMA
DP03_0088E
DP03_0088M
DP03_0136PE
DP03_0107PEA
DP03_0136PM
DP03_0082PM
DP03_0016MA
DP03_0082PE
DP03_0089E
DP03_0089M
DP03_0086E
DP03_0016PEA
DP03_0086M
DP03_0085M
DP03_0078EA
DP03_0087E
DP03_0110EA
DP03_0087M
DP03_0084E
DP03_0083M
DP03_0120PEA
DP03_0016PM
DP03_0085E
DP03_0136MA
DP03_0084M
DP03_0068PMA
DP03_0016PE
DP03_0110PE
DP03_0094E
DP03_0110PM
DP03_0093M
DP03_0094M
DP03_0065EA
DP03_0107PMA
DP03_0092E
DP03_0132PEA
DP03_0091M
DP03_0084PMA
DP03_0093E
DP03_0092M
DP03_0090E
DP03_0078MA
DP03_0110MA
DP03_0123PE
DP03_0072PMA
DP03_0059PMA
DP03_0123PM
DP03_0091E
DP03_0090M
DP03_0052EA
DP03_0120PMA
DP03_0060PMA
DP03_0078PM
DP03_0065MA
DP03_0078PE
DP03_0056PEA
DP03_0132PMA
DP03_0009M
DP03_0009E
DP03_0017EA
DP03_0070EA
DP03_0137E
DP03_0137M
DP03_0065PM
DP03_0106EA
DP03_0052MA
DP03_0065PE
DP03_0083EA
DP03_0014E
DP03_0083MA
DP03_0014M
DP03_0124EA
DP03_0015E
DP03_0106P

DP04_0064M
DP04_0064E
DP04_0072MA
DP04_0108PE
DP04_0094PEA
DP04_0108PM
DP04_0063M
DP04_0139PE
DP04_0126MA
DP04_0017PEA
DP04_0139PM
DP04_0141PMA
DP04_0069PMA
DP04_0070PMA
DP04_0005PEA
DP04_0024EA
DP04_0061E
DP04_0072PE
DP04_0100EA
DP04_0037EA
DP04_0060M
DP04_0090EA
DP04_0082PMA
DP04_0060E
DP04_0019PM
DP04_0139MA
DP04_0019PE
DP04_0072PM
DP04_0059M
DP04_0059E
DP04_0099PMA
DP04_0113MA
DP04_0058M
DP04_0090MA
DP04_0058E
DP04_0126PE
DP04_0126PM
DP04_0068EA
DP04_0123PMA
DP04_0055M
DP04_0090PE
DP04_0055E
DP04_0055EA
DP04_0054M
DP04_0037PM
DP04_0135PEA
DP04_0054E
DP04_0037PE
DP04_0053M
DP04_0024MA
DP04_0090PM
DP04_0111PMA
DP04_0113PE
DP04_0100MA
DP04_0057M
DP04_0113PM
DP04_0057E
DP04_0037MA
DP04_0051PMA
DP04_0056M
DP04_0056E
DP04_0131EA
DP04_0104M
DP04_0104E
DP04_0109EA
DP04_0099EA
DP04_0024PM
DP04_0024PE
DP04_0103M
DP04_0099PEA
DP04_0103E
DP04_0047PEA
DP04_0068MA
DP04_0035PEA
DP04_0106M
DP04_0063PMA
DP04_0106E
DP04_0105M
DP04_0100PE
DP04_0105E
DP04_0100PM
DP04_0042EA
DP04_0075PMA
DP04_0131PE
DP

DP05_0052EA
DP05_0032PEA
DP05_0035M
DP05_0035E
DP05_0044PEA
DP05_0078MA
DP05_0034PM
DP05_0021MA
DP05_0034PE
DP05_0056PEA
DP05_0030E
DP05_0021PM
DP05_0021PE
DP05_0031E
DP05_0030M
DP05_0072PMA
DP05_0078PM
DP05_0007PMA
DP05_0078PE
DP05_0065MA
DP05_0059PMA
DP05_0065PM
DP05_0065PE
DP05_0052MA
DP05_0060PMA
DP05_0017EA
DP05_0070EA
DP05_0052PM
DP05_0052PE
DP05_0060PEA
DP05_0011PMA
DP05_0008MA
DP05_0074PM
DP05_0074PE
DP05_0023PMA
DP05_0008PM
DP05_0008PE
DP05_0061PM
DP05_0059PEA
DP05_0061PE
DP05_0026EA
DP05_0008M
DP05_0008E
DP05_0013EA
DP05_0009M
DP05_0009E
DP05_0035PMA
DP05_0039MA
DP05_0072PEA
DP05_0047PMA
DP05_0004M
DP05_0004E
DP05_0057EA
DP05_0011PEA
DP05_0005M
DP05_0005E
DP05_0006M
DP05_0023PEA
DP05_0006E
DP05_0007M
DP05_0039PM
DP05_0007E
DP05_0039PE
DP05_0026MA
DP05_0026PM
DP05_0026PE
DP05_0013MA
DP05_0044EA
DP05_0035PEA
DP05_0031EA
DP05_0047PEA
DP05_0063PMA
DP05_0057MA
DP05_0051PMA
DP05_0013PM
DP05_0013PE
DP05_0075EA
DP05_0057PM
DP05_0044MA
DP05_0057PE
DP05_0075PMA
DP05_0044PM
DP05_0044PE


In [32]:
print('No of variable codes whose labels are different in 2014 and 2019 : {}'.format(count_mismatched_labels))

No of variable codes whose labels are different in 2014 and 2019 : 3544


Out of the 4088 common variable codes between 2014 and 2019, 86.69% of the variable codes didn't have the same label for both years. This is based on an exact string comparison.

Next Things: Create a new mapping between variable codes and labels considering variable codes vs label differences between 2014 and 2019. 